Feature/8033 refactor observable pattern

ZexinPeng opened the pull request.

Commit:
feat: log function using observer pattern

Commit:
feat: log function using observer pattern

Commit:
refactor: LogEntryService

Commit:
feature : log function using observer pattern

Commit:
refactor: javadoc in log function

Commit:
refactor : javadoc

Commit:
Revert "refactor : javadoc"

This reverts commit 8f1372bc73d8804cd16149f83a2354949cfa87f0.

Commit:
refatctor : add javadoc

add javadoc in LogService

ZexinPeng committed:
Merge branch 'develop' into feature/8033-refactor-observable-pattern

ZexinPeng committed:
Delete .classpath

ZexinPeng committed:
Delete .gitignore

ZexinPeng committed:
Delete LogServiceTest.java

fzuray commented on src/warzone/src/warzone/model/GameContext.java:
```diff
@@ -67,6 +72,19 @@ public static GameContext getGameContext() {
+		}
+		return LOG_ENTRY_BUFFER;
+	}
+	
```
move the singleton to LogEntryBuffer class, and only add datamember in the GameContext

fzuray committed:
update the implementation of logging
Merge branch 'develop' into feature/8033-refactor-observable-pattern

Conflicts:
	src/warzone/src/warzone/state/MapEditor.java

fzuray committed:
fix error
